### PR TITLE
Implement map and touch callbacks

### PR DESCRIPTION
**Pull request by naftalibeder**
## Description

- Implements region and long-press handlers on iOS for v10.
- Adds new v10-only `cameraChanged` and `mapIdle` callbacks that better mirror the v10 native APIs.
- Listens for gesture events to pass user interaction info, adds `isAnimatingFromUserInteraction` prop denoting a map movement caused by a user interaction (even if the user is no longer interacting).
- Add example dedicated to gesture handlers.

cc @mfazekas 

## Checklist

- [X] I have tested this on a device/simulator for each compatible OS (iOS only)
- [X] I formatted JS and TS files with running `yarn lint:fix` in the root folder
- [X] I updated the documentation with running `yarn generate` in the root folder
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typings files (`index.d.ts`)
- [X] I added/ updated a sample (`/example`)

## Video

https://user-images.githubusercontent.com/2423291/164084873-7b8c7de5-0df9-4853-a0fc-bd65e28a3c4b.mov

### CODE REVIEW

**Review by mfazekas — APPROVED** [review]

File: ios/RCTMGL-v10/RCTMGLMapView.swift [hunk 1]
```diff
@@ -152,23 +150,31 @@ open class RCTMGLMapView : MapView {
...
-      let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap))
-      self.addGestureRecognizer(tapGesture)
-    */
-    mapView.gestures.singleTapGestureRecognizer.removeTarget( pointAnnotationManager.manager, action: nil)
-    mapView.gestures.singleTapGestureRecognizer.addTarget(self, action: #selector(doHandleTap(_:)))
+  @objc func setReactOnLongPress(_ value: @escaping RCTBubblingEventBlock) {
+    self.reactOnLongPress = value
+
+    let longPressGestureRecognizer = UILongPressGestureRecognizer(target: self, action: #selector(doHandleLongPress(_:)))
+    self.mapView.addGestureRecognizer(longPressGestureRecognizer)
```
Comment (mfazekas) [inline ios/RCTMGL-v10/RCTMGLMapView.swift:165]: Not sure what happens if onLongPress is changing, will be keep adding gesture recognisers.

Comment (naftalibeder) [reply ios/RCTMGL-v10/RCTMGLMapView.swift:165]: That's a good question - I'm pretty sure that the `setReact*` functions defined in `RCTMGLMapViewManager.m` are just run once when the native module is loaded - I will verify.

I guess it couldn't hurt to store the recognizer on the class and do a nil-check, but I don't want to add complexity if it's not needed.

Comment (mfazekas) [reply ios/RCTMGL-v10/RCTMGLMapView.swift:165]: Ok I guess it's fine as it's is for now, we can improve later

Comment (mfazekas) [reply ios/RCTMGL-v10/RCTMGLMapView.swift:165]: 👍

**Pull request merged**